auth/api/iam/session.go

```diff
@@ -39,17 +39,76 @@ type OAuthSession struct {
 	ClientState            string
 	SessionID              string
 	RedirectURI            string
-	ServerState            ServerState
 	ResponseType           string
 	PKCEParams             PKCEParams
 	PresentationDefinition PresentationDefinition
 	VerifierDID            *did.DID
+	OpenID4VPVerifier      *OpenID4VPVerifier
+}
 
-	// TODO use these 2 fields to track if all OpenID4VP flows have been concluded
-	// PresentationSubmissions tracks which PresentationSubmissions have been submitted through OpenID4VP
-	PresentationSubmissions map[string]pe.PresentationSubmission
+// OpenID4VPVerifier tracks the verifier's state of multiple OpenID4VP flows that all relate to a single OAuthSession (e.g. Authorization Code flow).
+// It's a sort of state machine with 2 states.
+type OpenID4VPVerifier struct {
```

Comment:
It would be a good idea to start adding JSON tags to these session storage structs. Non in-memory storage would benefit from it.